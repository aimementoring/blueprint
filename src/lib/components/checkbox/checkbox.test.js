import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from './checkbox';

describe('Checkbox', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(
        <Checkbox
          elementClassName="classname"
          onChangeFunction={() => {}}
          placeholder="Accept terms and conditions"
          name="checkbox"
          customId="checkbox"
          value={true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
