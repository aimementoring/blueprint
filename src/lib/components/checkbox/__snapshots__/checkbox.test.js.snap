// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders properly 1`] = `
<div
  className="theme-purple wrapper"
>
  <div
    className=" customCheckbox customCheckboxDefault"
    onClick={[Function]}
  >
    <input
      name="checkbox"
      type="hidden"
      value="no"
    />
    <input
      checked={true}
      className="false"
      id="checkbox"
      name="checkbox"
      readOnly={true}
      type="checkbox"
      value="yes"
    />
    <label
      htmlFor="checkbox"
    >
      Accept terms and conditions
    </label>
  </div>
</div>
`;

exports[`Checkbox renders properly when value is invalid 1`] = `
<div
  className="theme-purple wrapper"
>
  <div
    className=" customCheckbox customCheckboxDefault"
    onClick={[Function]}
  >
    <input
      name="checkbox"
      type="hidden"
      value="no"
    />
    <input
      checked={true}
      className="error"
      id="checkbox"
      name="checkbox"
      readOnly={true}
      type="checkbox"
      value="yes"
    />
    <label
      htmlFor="checkbox"
    >
      Accept terms and conditions
    </label>
  </div>
</div>
`;
